maven-plugin/rcptt-maven-util/src/main/java/org/eclipse/rcptt/maven/util/PlatformUtil.java

```diff
@@ -19,23 +19,33 @@ public class PlatformUtil {
 	 * @return The classifier of current platform in form of os.ws.arch
 	 */
 	public static String getEclipseClassifier() {
-		return String.format("%s.%s", getOS().classifier, getArch());
+		return String.format("%s.%s", getOS().classifier, getArch()); //$NON-NLS-1$
 	}
 
 	public static String getArch() {
-		String result = System.getProperty("os.arch");
-		if (result.contains("64") || getOS() == OS.MACOSX) {
-			return "x86_64";
+		String result = System.getProperty("os.arch"); //$NON-NLS-1$
+		switch (result.toLowerCase()) {
+		case "i386": //$NON-NLS-1$
+		case "i486": //$NON-NLS-1$
+		case "i586": //$NON-NLS-1$
+		case "i686": //$NON-NLS-1$
+		case "pentium": //$NON-NLS-1$
+			return "x86"; //$NON-NLS-1$
+		case "amd64": //$NON-NLS-1$
+		case "em64t": //$NON-NLS-1$
+		case "universal": //$NON-NLS-1$
+			return "x86_64"; //$NON-NLS-1$
+		default:
+			return result;
```

Comment:
Why do we return an arbitrary string here? Should not this method guarantee a fixed set of possible return values?

Reply:
I am happy to apply whatever change you recommend.  From my perspective, the problem is that this method would previously return a fixed set of values which could be the wrong value.  In particular, it was hard coded to always return either "x86" or "x86_64".  I could add explicit identity mappings like ```case "x86": return "x86"```, ```case "x86_64": return "x86_64"```, and ```case "aarch64": return "aarch64"``` but then the question is what the default case should return?  We could hard code the default case to something but it would likely be incorrect if we were to get there.  In contrast, if we return original "os.arch" string (like what I suggest) then this might actually be correct which is indeed the case for aarch64.

Here is some additional background information I was able to dig up about this topic:

The source for this string seems like it likely comes from the [java_props_md.c for Windows](https://github.com/openjdk/jdk/blob/master/src/java.base/windows/native/libjava/java_props_md.c) or the [java_props_md.c for Unix](https://github.com/openjdk/jdk/blob/master/src/java.base/unix/native/libjava/java_props_md.c).  For Windows, "GetJavaProperties" sets "sprops.os_arch" to "amd64", "x86", "aarch64", or "unknown".  For Unix, GetJavaProperties seems to get it from an externally defined ARCHPROPNAME macro value which is vague/uncertain.

How do other things translate "os.arch" into a better OSGi-friendly arch value?  There is [org.eclipse.equinox.launcher.Main.getArch()](https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java) which returns an externally defined "osgi.arch" but falls back on "os.arch", mapping "amd64" to "x86_64" but does fall back on returning original "os.arch" value.  This opens the question as to whether there are other mappings or if "amd64" to "x86_64" is the only mapping we need or care about.  The [org.apache.commons.lang3.ArchUtils](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/ArchUtils.java) includes some additional mappings including some old 32-bit mappings like "i386", "i486", "i586", "i686", and "pentium", for example.  So, at least coding in such mappings seems possibly safer than incorrectly falling back on "x86" like the code was previously doing to handle such cases.

Reply:
RCPTT suffered quite a lot from incorrect results returned in similar methods. I assume (but did not check), that clients expect this method to return OSGI-compatible value, but only one of x86, x86_64, aarch64. As Eclipse no longer supports x86, we can drop it. So I suggest to return either x86_64, aarch64 or throw an exception.
Yes, this will cause regressions, but multi-stage debugging and log analysis required by occasional invalid values is no longer acceptable.

Reply:
Explicit conversions are fine. My object was for the default value.
IllegalStateException is fine.